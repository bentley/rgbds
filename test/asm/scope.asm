SECTION "test", ROM0

	assert !STRLEN("{__SCOPE__}")

SomeFunction:
	println __FILE__, ":{d:__LINE__}:{__SCOPE__}: 2 + 2 = ", 2+2

verify: MACRO
	assert !STRCMP("{__SCOPE__}", "\1")
	assert __SCOPE__ == \1
ENDM

Alpha:
	verify Alpha
.local
	verify Alpha

Beta::
.local::
	verify Beta

S EQUS """
Gamma:
	verify Gamma
"""
	S
	PURGE S
	verify Gamma
