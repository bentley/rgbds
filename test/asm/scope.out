scope.asm:6:SomeFunction: 2 + 2 = $4
